components/core/src/ffi/ir_stream/byteswap.hpp

```diff
@@ -0,0 +1,14 @@
+#ifndef FFI_IR_STREAM_BYTESWAP_HPP
+#define FFI_IR_STREAM_BYTESWAP_HPP
+
+// C standard libraries
+#ifdef __APPLE__
```

Comment:
Nit: Can we change this to a more generic detection 
```
#ifndef _BYTESWAP_H
#define _BYTESWAP_H
#endif
```
byteswap seems to be GNU specific header and other linux flavors such as BSD may lack this header as well.